### PR TITLE
feat(ebike): replace charging point scan with distance-based e-bike range alerts

### PR DESCRIPTION
**Pull request by vincentchalamon**
## Summary

- Replace the unreliable OSM `amenity=charging_station` + `bicycle=yes` scan with a distance-based **EbikeRangeAnalyzer** that triggers when stage distance exceeds effective e-bike range (`max(0, 80km - D+/25)`)
- Add user-togglable `ebikeMode` option (default `false`) to `TripRequest` DTO, Zustand store, and `PacingSettings` UI
- Remove entire `CheckChargingPoints` infrastructure (message, handler, test, Overpass query, computation enum, Mercure event type)

## Test plan

- [x] `make qa` — PHPStan L9, CS-Fixer, Rector, ESLint, Prettier, TS check all pass
- [x] `make phpunit` — 346 tests (including 7 `EbikeRangeAnalyzerTest` cases), 945 assertions
- [x] `make typegen` — `schema.d.ts` regenerated with `ebikeMode` field
- [x] No remaining references to `charging_point` anywhere in the codebase
- [ ] `make test-e2e` — E2E tests (no contract change for `terrain_alerts`)

## Auto-critique

- [x] Code respects project architecture (stateless backend, local-first frontend, DTO contract)
- [x] SOLID principles followed — `EbikeRangeAnalyzer` is a single-purpose analyzer auto-discovered via tagged services
- [x] No dead code left (all `charging_point` references removed, unused `lat`/`lon` removed from Alert)
- [x] No lingering TODO/FIXME comments
- [x] Translations updated (en/fr) for both backend alerts and frontend UI
- [x] TypeScript types regenerated from OpenAPI spec
- [x] Effective range clamped to `max(0, ...)` to avoid negative range display on extreme elevation
- [x] Shared `patchPacingSettings` helper eliminates PATCH body duplication

🤖 Generated with [Claude Code](https://claude.ai/code)

## Claude Review

**Commit reviewed:** 66596e8867d1679032c88cf662748e8ee2529496

Clean, well-structured final state. The three fix commits since the last review correctly addressed all outstanding findings: `checkContinuity` dead code removed from `RecalculateStages`, the Radix UI `Switch` component replaced the raw checkbox, and the missing-context edge-case test was added. No new issues found.

One previously open bot-authored nitpick on `patchPacingSettings` (`clearStages: boolean` flag argument) remains unaddressed; it is non-blocking and the trade-off is reasonable.

### Resolved threads

No new threads resolved — the 9 previously resolved threads remain resolved. The 1 open bot-authored thread (`pwa/src/hooks/use-trip-planner.ts:234`, `clearStages` flag argument) is not fixed in this push and remains open.

### Review checklist

- [x] Code respects the project architecture (stateless backend, local-first frontend, DTO contract)
- [x] SOLID principles and Law of Demeter followed (`EbikeRangeAnalyzer` SRP; `ComputationDependencyResolver` OCP via `PARAMETER_DEPENDENCIES` map)
- [x] Design patterns used where appropriate (`StageAnalyzerInterface` / Chain of Responsibility; `patchPacingSettings` pure API helper; Radix UI `Switch` design-system component)
- [x] Tests cover new/changed cases (8 unit tests for `EbikeRangeAnalyzer` incl. clamped-range, missing-key, and priority; dependency resolver test for `ebikeMode`)
- [~] E2E tests — no E2E test for the new e-bike mode toggle UI; PR body explicitly marks this unchecked (no contract change to `terrain_alerts`, acceptable)
- [x] Documentation is up to date (translations en/fr, Zod schema, OpenAPI-regenerated TypeScript types)
- [x] Dependent tickets accounted for

### No inline comments.

Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: api/src/MessageHandler/AnalyzeTerrainHandler.php [hunk 1]
```diff
@@ -36,8 +36,10 @@ public function __invoke(AnalyzeTerrain $message): void
         }
 
         $locale = $this->tripStateManager->getLocale($tripId) ?? 'en';
+        $request = $this->tripStateManager->getRequest($tripId);
+        $ebikeMode = $request instanceof \App\ApiResource\TripRequest && $request->ebikeMode;
```
[the remark this answers is not in the document]
Comment (vincentchalamon) [reply api/src/MessageHandler/AnalyzeTerrainHandler.php]: Fixed in f5bd52e. Replaced with `(bool) ($this->tripStateManager->getRequest($tripId)?->ebikeMode)` — the nullsafe handles `getRequest()` returning `null`, and the `(bool)` cast converts `null` to `false`. PHPStan L9 passes (the `?->` + `??` combo was flagged because `ebikeMode` is `bool`, never `null`).
